### PR TITLE
fix: optimize spaces helpers 

### PR DESCRIPTION
**Pull request by wa0x6e**
Use the new `vote_count`, `proposal_count` and `follower_count` columns from the spaces table to retrieve total number of items per space, instead of using a `COUNT(id) as COUNT` select.

Also some refactoring to add types, and cleaning.

The whole script now run in 40 seconds (instead of > 10 minutes), with 95% of the time spend on loading all spaces.

### How to test

1. Connect your snapshot UI to a hub with running branch
2. Front page should load the same spaces, in the same order

### CODE REVIEW

File: src/helpers/spaces.ts [hunk 1]
```diff
@@ -134,28 +134,40 @@ async function loadSpaces() {
...
 
 async function getFollowers() {
   const query = `
-    SELECT space, COUNT(id) as count,
-    count(IF(created > (UNIX_TIMESTAMP() - 604800), 1, NULL)) as count_7d
-    FROM follows GROUP BY space
+    SELECT
+      space,
+      spaces.follower_count AS count,
+      COUNT(IF(created > (UNIX_TIMESTAMP() - 604800), 1, NULL)) AS count_7d
```
Comment (ChaituVR) [inline src/helpers/spaces.ts:167]: I get this error `Column 'created' in field list is ambiguous` , I think it should be `follows.created`, will add a fix

File: src/helpers/spaces.ts [hunk 2]
```diff
@@ -134,28 +134,40 @@ async function loadSpaces() {
...
-    SELECT space, COUNT(id) as count,
-    count(IF(created > (UNIX_TIMESTAMP() - 604800), 1, NULL)) as count_7d
-    FROM votes GROUP BY space
+    SELECT
+      space,
+      spaces.vote_count as count,
+      COUNT(IF(v.created > (UNIX_TIMESTAMP() - 604800), 1, NULL)) AS count_7d
+    FROM votes v
+    JOIN spaces ON spaces.id = space
+    GROUP BY space
```
Comment (ChaituVR) [inline src/helpers/spaces.ts:157]: This one is taking more than 20 mins (still running) but the previous one took around 12.5 mins

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Comment by ChaituVR** [comment]
I think solution should be to ignore total counts here in these queries, but use space counts directly on `mapSpaces` in `src/helpers/spaces.ts`

File: src/helpers/spaces.ts (hunk 2 above)
Comment (ChaituVR) [reply src/helpers/spaces.ts:157]: It took 24 mins

Comment (wa0x6e) [reply src/helpers/spaces.ts:157]: How can a simple read on a column `vote_count` be slower than a `COUNT(id)` ?

Comment (wa0x6e) [reply src/helpers/spaces.ts:157]: Maybe the slowness is not related to the update, and the whole query itself does not take same time on each run ? (sometime 12min, sometime 20min, sometime 30min) ?

File: src/helpers/spaces.ts [hunk 3]
```diff
@@ -134,28 +134,40 @@ async function loadSpaces() {
...
-    SELECT space, COUNT(id) as count,
-    count(IF(created > (UNIX_TIMESTAMP() - 604800), 1, NULL)) as count_7d
-    FROM follows GROUP BY space
+    SELECT
+      space,
+      spaces.follower_count AS count,
+      COUNT(IF(f.created > (UNIX_TIMESTAMP() - 604800), 1, NULL)) AS count_7d
+    FROM follows f
+    JOIN spaces ON spaces.id = space
+    GROUP BY space
```
[the remark this answers is not in the document]
Comment (wa0x6e) [reply src/helpers/spaces.ts:170]: Let me try an `explain`, it's very weird

[the remark this answers is not in the document]
Comment (wa0x6e) [reply src/helpers/spaces.ts:170]: Ah, `JOIN` was the expensive part. I may have found a better query, which is almost instantaneous 😎😎

File: src/helpers/spaces.ts (hunk 2 above)
Comment (ChaituVR) [reply src/helpers/spaces.ts:157]: I think slowness could be because of JOIN spaces ON spaces.id = space, adding spaces to all votes

Comment (ChaituVR) [reply src/helpers/spaces.ts:157]: we already have all space data in `mapSpaces` so we don't need to get this data here, we can just get `7d` counts here and we should be good

Comment (wa0x6e) [reply src/helpers/spaces.ts:157]: Yes, i found a better query, everything will be done in less than 10s

**Review by ChaituVR — APPROVED** [review]
tAck 👍

**Pull request merged**